src/gl/shader.cpp

```diff
@@ -44,23 +44,19 @@ bool find_id(const std::string& program, const char* id) {
     return std::strstr(program.c_str(), id) != 0;
 }
 
-bool Shader::load(const std::string& _fragmentSrc, const std::string& _vertexSrc, bool verbose) {
+bool Shader::load(const std::string* _fragmentPath, const std::string& _fragmentSrc, const std::string* _vertexPath, const std::string& _vertexSrc, bool verbose) {
     std::chrono::time_point<std::chrono::steady_clock> start_time, end_time;
     start_time = std::chrono::steady_clock::now();
 
-    m_vertexShader = compileShader(_vertexSrc, GL_VERTEX_SHADER);
+    m_vertexShader = compileShader(_vertexPath, _vertexSrc, GL_VERTEX_SHADER);
```

Comment:
If `#include` is used, then (currently) line numbers will be incorrect. And I have no way to report the names of included files correctly in warning/error messages. That is because the GLSL standard doesn't permit you to put a pathname into a `#line` directive. So it's hard to fix this problem.

The _vertexPath and _shaderPath arguments of compileShader are the pathnames of the top level *.vert and *.frag files, which are globally unique. These pathnames are used only when printing messages.

I haven't tested `#include` yet. I will.